src/aspire/basis/fb.py

```diff
@@ -0,0 +1,55 @@
+import logging
+
+import numpy as np
+
+from aspire.basis.basis_utils import num_besselj_zeros
+from aspire.utils.matlab_compat import m_reshape
+
+logger = logging.getLogger(__name__)
+
+
+class FBBasisMixin(object):
+    """
+    FBBasisMixin is a mixin implementing methods specific to Fourier-Bessel expansions,
+    to be inherited by Fourier-Bessel subclasses of Basis.
+    """
+
+    def _getfbzeros(self):
+        """
+        Generate zeros of Bessel functions
+        """
+        # get upper_bound of zeros of Bessel functions
+        upper_bound = min(self.ell_max + 1, 2 * self.nres + 1)
+
+        # List of number of zeros
+        n = []
+        # List of zero values (each entry is an ndarray; all of possibly different lengths)
+        zeros = []
+
+        # generate zeros of Bessel functions for each ell
+        for ell in range(upper_bound):
+            # for each ell, num_besselj_zeros returns the zeros of the
+            # order ell Bessel function which are less than 2*pi*c*R = nres*pi/2,
+            # the truncation rule for the Fourier-Bessel expansion
+            _n, _zeros = num_besselj_zeros(
+                ell + (self.ndim - 2) / 2, self.nres * np.pi / 2
```

Comment:
Better to add an `if` statement here to select between `self.ndim == 2` and `self.ndim == 3`. Don't think this generalizes beyond dimensions 2 and 3.

Reply:
But it still gives the wrong impression. If `ndim == 2`, we have `ell`, otherwise we have `ell + 1/2`. We don't want to imply a linear relationship between the number of dimensions and the `ell` if such a thing does not exist.

Reply:
Okay, I made  the rule more explicit